vars/shellCheck.groovy

```diff
@@ -0,0 +1,39 @@
+/**
+ * shellcheck for jenkins-pipelines https://github.com/koalaman/shellcheck
+ *
+ */
+package com.cloudogu.ces.dogubuildlib
+
+/**
+ * run shellcheck with a custom fileList
+ * sample input "test1.sh", "test2.sh"
+ *
+ */
+def call(fileList) {
+    executeWithDocker(fileList)
+}
+
+/**
+ * run shellcheck on every .sh file inside the project folder
+ * note: it ignores ./ecosystem folder for usage with ecosystem instances
+ *
+ */
+def call() {
+    try{
+        def fileList = sh (script: 'find . -path ./ecosystem -prune -o -type f -regex .*\\.sh -print', returnStdout: true)
+        fileList='"'+fileList.trim().replaceAll('\n','" "')+'"'
+        executeWithDocker(fileList)
+    }catch(error){
```

Comment:
This is useful when we add additional behavior on errors. E.g a parameter to configure if the build should break on shellcheck issues or not. This feature is not implemented because of time constraints. Removed try-catch for now https://github.com/cloudogu/dogu-build-lib/pull/5/commits/452f8fdd2d802e1185a4c848cea6ce3d430d086a